### PR TITLE
Default PCH to on

### LINKED ISSUE #567: Default USE_PCH to ON and always use that for CI
> I really like this PR and I am actually in favor of setting USE_PCH to ON as default value in the CMakeLists.txt files and to enable it for all the travis builds too.

_Originally posted by @bourtemb in https://github.com/tango-controls/cppTango/pull/526#issuecomment-500917259_

### PR DESCRIPTION
**Pull request by t-b**
Close #567.

### CODE REVIEW

**Comment by mliszcz** [comment]
Looks good to me.

File: .travis.yml [hunk 1]
```diff
@@ -3,13 +3,22 @@ dist: xenial
...
   matrix:
-    - OS_TYPE=gcc-latest CMAKE_BUILD_TYPE=Debug   SONAR_SCANNER=OFF COVERALLS=OFF RUN_TESTS=OFF WARNINGS_AS_ERRORS=ON STOCK_CPPZMQ=ON TANGO_USE_USING_NAMESPACE=OFF USE_PCH=ON
-    - OS_TYPE=debian10 CMAKE_BUILD_TYPE=Debug   SONAR_SCANNER=OFF COVERALLS=OFF RUN_TESTS=ON STOCK_CPPZMQ=ON TANGO_USE_USING_NAMESPACE=OFF USE_PCH=ON
-    - OS_TYPE=debian10 CMAKE_BUILD_TYPE=Release SONAR_SCANNER=OFF COVERALLS=OFF RUN_TESTS=ON STOCK_CPPZMQ=ON
-    - OS_TYPE=debian9  CMAKE_BUILD_TYPE=Debug   SONAR_SCANNER=OFF COVERALLS=OFF RUN_TESTS=ON STOCK_CPPZMQ=ON
-    - OS_TYPE=debian8  CMAKE_BUILD_TYPE=Debug   SONAR_SCANNER=ON  COVERALLS=ON  RUN_TESTS=ON STOCK_CPPZMQ=OFF
-    - OS_TYPE=debian7  CMAKE_BUILD_TYPE=Debug   SONAR_SCANNER=OFF COVERALLS=OFF RUN_TESTS=ON STOCK_CPPZMQ=ON
+    - OS_TYPE=gcc-latest RUN_TESTS=OFF WARNINGS_AS_ERRORS=ON TANGO_USE_USING_NAMESPACE=OFF
+    - OS_TYPE=debian10
+    - OS_TYPE=debian10 USE_PCH=OFF RUN_TESTS=OFF
```
Comment (bourtemb) [inline .travis.yml]: We don't run the tests in this case.
We are confident enough the behaviour is (and will stay) the same as when USE_PCH=ON?

Comment (mliszcz) [reply .travis.yml]: I'm not against enabling tests here, but the important fact is that we still *compile* the tests. If the behavior is different and the compilation is not failing, then it's rather a compiler issue.

Comment (t-b) [reply .travis.yml]: Yes I'm confident that just compiling code and tests is enough.

**Comment by bourtemb** [comment]
This PR disabled SONAR_SCANNER and COVERALLS which used to be ON when building on Debian 8.
I don't remember whether we discussed this point or if we missed that point when reviewing this PR?

**Comment by t-b** [comment]
@bourtemb Good point. I think we want

```
OS_TYPE=debian10 USE_PCH=OFF RUN_TESTS=OFF SONAR_SCANNER=ON
```

or?

**Comment by bourtemb** [comment]
> 
> 
> @bourtemb Good point. I think we want
> 
> ```
> OS_TYPE=debian10 USE_PCH=OFF RUN_TESTS=OFF SONAR_SCANNER=ON
> ```
> 
> or?

and COVERALLS=ON too

**Comment by t-b** [comment]
@bourtemb Yes, but let's go with readding it to debian8 as @mliszcz did in #568.

**Comment by mliszcz** [comment]
Hi @t-b and @bourtemb 

Do you have any experience with sonar and coveralls setup? Below are logs from Debian 8 job:

https://travis-ci.org/tango-controls/cppTango/jobs/640430653?utm_medium=notification&utm_source=github_status

Coveralls:
```
[  0%] Built target coveralls_generate
make -f CMakeFiles/coveralls_upload.dir/build.make CMakeFiles/coveralls_upload.dir/depend
make[3]: Entering directory '/home/tango/src/build'
cd /home/tango/src/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/tango/src /home/tango/src /home/tango/src/build /home/tango/src/build /home/tango/src/build/CMakeFiles/coveralls_upload.dir/DependInfo.cmake --color=
Scanning dependencies of target coveralls_upload
make[3]: Leaving directory '/home/tango/src/build'
make -f CMakeFiles/coveralls_upload.dir/build.make CMakeFiles/coveralls_upload.dir/build
make[3]: Entering directory '/home/tango/src/build'
/usr/bin/cmake -E cmake_progress_report /home/tango/src/build/CMakeFiles 34
[100%] Uploading coveralls output...
/usr/bin/curl -S -F json_file=@/home/tango/src/build/coveralls.json https://coveralls.io/api/v1/jobs
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
 15 1031k    0     0   15  160k      0  1260k --:--:-- --:--:-- --:--:-- 1261k
100 1031k    0    72  100 1031k     54   774k  0:00:01  0:00:01 --:--:--  773k
100 1031k    0    72  100 1031k     54   774k  0:00:01  0:00:01 --:--:--  773k
{"message":"Couldn't find a repository matching this job.","error":true}make[3]: Leaving directory '/home/tango/src/build'
```

Coverate is not visible: https://coveralls.io/github/tango-controls/cppTango

Sonar:
```
[0K$ test ${SONAR_SCANNER} = "ON" && .travis/sonar.sh
Using SONAR_PROPERTIES=-Dsonar.pullrequest.base=tango-9-lts -Dsonar.pullrequest.branch=reset-database-and-ds-for-each-test -Dsonar.pullrequest.key=640
.travis/sonar.sh: line 11: sonar-scanner: command not found
```

I'm trying to find some hints in commit history but with no luck so far.

**Comment by t-b** [comment]
@mliszcz Maybe coverage is only computed for `tango-9-lts`? In any case I would tackle that in a different PR.

**Pull request merged**